src/var/task/handler.py

```diff
@@ -0,0 +1,205 @@
+import json
+import os
+import subprocess
+from datetime import datetime
+
+import boto3
+import botocore.exceptions
+
+s3_client = boto3.client("s3")
+sns_client = boto3.client("sns")
+scan_time = datetime.now().isoformat()
+
+
+def run_command(command):
+    result = subprocess.run(  # pylint: disable=subprocess-run-check
+        command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+    return (
+        result.returncode,
+        result.stdout.decode("utf-8"),
+        result.stderr.decode("utf-8"),
+    )
+
+
+def definition_upload():
+    try:
+        # Create the directory to store the definitions
+        os.makedirs("/tmp/clamav/database", exist_ok=True)
+
+        # Pull the latest definitions
+        user_id = run_command("id --user")[1].strip()
+        clamav_config = os.environ.get("LAMBDA_TASK_ROOT", "") + "/freshclam.conf"
+        run_command(
+            f'freshclam --no-warnings --user {user_id} --config-file="{clamav_config}"'
+        )
+
+        # Archive the definitions
+        run_command(
+            "tar --create --gzip --verbose --file=/tmp/clamav/clamav.tar.gz -C /tmp/clamav/database ."
+        )
+
+        # Upload the definitions to S3
+        bucket_name = os.environ.get("CLAMAV_DEFINITON_BUCKET_NAME")
+        if not bucket_name:
+            raise ValueError(
+                "CLAMAV_DEFINITON_BUCKET_NAME environment variable not set."
+            )
+        s3_client.upload_file("/tmp/clamav/clamav.tar.gz", bucket_name, "clamav.tar.gz")
+    except botocore.exceptions.ClientError as e:
+        print(f"Failed to upload ClamAV definitions: {e}")
+
+
+def definition_download():
+    try:
+        # Create the directory to store the definitions
+        os.makedirs("/tmp/clamav/database", exist_ok=True)
+
+        # Download the definitions from S3
+        bucket_name = os.environ.get("CLAMAV_DEFINITON_BUCKET_NAME")
+        if not bucket_name:
+            raise ValueError(
+                "CLAMAV_DEFINITON_BUCKET_NAME environment variable not set."
+            )
+        s3_client.download_file(
+            bucket_name, "clamav.tar.gz", "/tmp/clamav/clamav.tar.gz"
+        )
+        print("Successfully downloaded ClamAV definitions from S3.")
+
+        # Extract the definitions
+        run_command(
+            "tar --extract --gzip --verbose --file=/tmp/clamav/clamav.tar.gz -C /tmp/clamav/database"
+        )
+        print("Successfully extracted ClamAV definitions.")
+    except botocore.exceptions.ClientError as e:
+        print(f"Failed to download or extract ClamAV definitions: {e}")
+
+
+def scan(event):
+    # event_json = json.loads(event_data)
+    object_key = event["Records"][0]["s3"]["object"]["key"]
```

Comment:
Potential for `KeyError` here if the stucture of the event json is not what you expect. But I presume this is expected/ok? As there is error handling in `handler`

Reply:
But if you wanted to, you could wrap lines 80-81 in a `try/except` to catch `KeyError` and `IndexError` and add some logging and raise an exception with a descriptive message, similar to whatg you do with getting the `landing_bucket_name`